File: app/src/main/java/com/yao/zhihudaily/tool/SimpleDividerDecoration.java
package com.yao.zhihudaily.tool;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.Rect;
import android.view.View;

import com.yao.zhihudaily.R;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

/**
 *
 * @author Yao
 * @date 2017/3/29
 */
public class SimpleDividerDecoration extends RecyclerView.ItemDecoration {

    private int dividerHeight;
    private Paint dividerPaint;

    public SimpleDividerDecoration(Context context) {
        dividerPaint = new Paint();
        dividerPaint.setColor(context.getResources().getColor(R.color.item_decoration));
        dividerHeight = context.getResources().getDimensionPixelSize(R.dimen.divider_height);
    }


    @Override
    public void getItemOffsets(@NonNull Rect outRect, @NonNull View view, @NonNull RecyclerView parent, @NonNull RecyclerView.State state) {
        super.getItemOffsets(outRect, view, parent, state);
        outRect.bottom = dividerHeight;
    }

    @Override
    public void onDraw(@NonNull Canvas canvas, RecyclerView parent, @NonNull RecyclerView.State state) {
        int childCount = parent.getChildCount();
        int left = parent.getPaddingStart();
        int right = parent.getWidth() - parent.getPaddingEnd();

        for (int i = 0; i < childCount - 1; i++) {
            View view = parent.getChildAt(i);
            float top = view.getBottom();
            float bottom = view.getBottom() + dividerHeight;
            canvas.drawRect(left, top, right, bottom, dividerPaint);
        }
    }
}

File: app/src/main/java/com/yao/zhihudaily/tool/SimpleDividerDecoration.kt
package com.yao.zhihudaily.tool

import android.content.Context
import android.graphics.Canvas
import android.graphics.Paint
import android.graphics.Rect
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.yao.zhihudaily.R

/**
 *
 * @author Yao
 * @date 2017/3/29
 */
class SimpleDividerDecoration(context: Context) : RecyclerView.ItemDecoration() {

    private val dividerHeight: Int
    private val dividerPaint: Paint

    init {
        dividerPaint = Paint()
        dividerPaint.color = context.resources.getColor(R.color.item_decoration)
        dividerHeight = context.resources.getDimensionPixelSize(R.dimen.divider_height)
    }


    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
        super.getItemOffsets(outRect, view, parent, state)
        outRect.bottom = dividerHeight
    }

    override fun onDraw(canvas: Canvas, parent: RecyclerView, state: RecyclerView.State) {
        val childCount = parent.childCount
        val left = parent.paddingStart
        val right = parent.width - parent.paddingEnd

        for (i in 0 until childCount - 1) {
            val view = parent.getChildAt(i)
            val top = view.bottom.toFloat()
            val bottom = (view.bottom + dividerHeight).toFloat()
            canvas.drawRect(left.toFloat(), top, right.toFloat(), bottom, dividerPaint)
        }
    }
}
